src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -5,40 +5,51 @@
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class StreamStudy {
 
-	public static long countWords() throws IOException {
-		String contents = new String(Files.readAllBytes(Paths
-				.get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
-		List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
-
-		long count = 0;
-		for (String w : words) {
-		  if (w.length() > 12) count++;  
-		}
-		return count;
-	}
-	
-	public static void printLongestWordTop100() throws IOException {
-		String contents = new String(Files.readAllBytes(Paths
-				.get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
-		List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
-		
-		// TODO 이 부분에 구현한다.
-	}
-
-	public static List<Integer> doubleNumbers(List<Integer> numbers) {
-		return numbers.stream().map(x -> 2 * x).collect(Collectors.toList());
-	}
-
-	public static long sumAll(List<Integer> numbers) {
-		return numbers.stream().reduce(0, (x, y) -> x + y);
-	}
-
-	public static long sumOverThreeAndDouble(List<Integer> numbers) {
-		return 0;
-	}
+    public static long countWords() throws IOException {
+        String contents = new String(Files.readAllBytes(Paths
+                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
+        List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
+
+        return words.stream()
+				.filter(word -> word.length() > 12)
+				.count();
+    }
+
+    public static void printLongestWordTop100() throws IOException {
+        String contents = new String(Files.readAllBytes(Paths
+                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
+        List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
+
+        words.stream()
+                .distinct()
+                .filter(word -> word.length() > 12)
+                .sorted(Comparator.comparing(String::length))
+                .limit(100)
+                .map(String::toLowerCase)
+                .forEach(System.out::println);
+
+
+        // TODO 이 부분에 구현한다.
+    }
+
+    public static List<Integer> doubleNumbers(List<Integer> numbers) {
+        return numbers.stream().map(x -> 2 * x).collect(Collectors.toList());
+    }
+
+    public static long sumAll(List<Integer> numbers) {
+        return numbers.stream().reduce(0, (x, y) -> x + y);
+    }
+
+    public static long sumOverThreeAndDouble(List<Integer> numbers) {
+        return numbers.stream()
+                .filter(i -> i > 3)
+                .mapToInt(i -> i * 2)
+                .reduce(0, (i1, i2) -> i1 + i2);
```

Comment:
이 부분도 Method Reference를 활용할 수 있겠네요

```suggestion
                .reduce(0, Integer::sum);
```